driver/compile_common.ml

```diff
@@ -42,11 +46,31 @@ let with_info ~native ~tool_name ~source_file ~output_prefix ~dump_ext k =
     output_prefix;
     env;
     source_file;
+    human_source_file=source_file;
     ppf_dump;
     tool_name;
     native;
   }
 
+
+(** Recompute human source path using line directives *)
+let human_source_path proj ast info =
+  let location loc item =
+    let item_loc = proj item in
+    if item_loc.Location.loc_ghost then loc
+      else
+        Some item_loc.Location.loc_start.Lexing.pos_fname
+  in
+  match List.fold_left location None ast with
+  | None -> info
+  | Some source -> { info with human_source_file = source }
```

Comment:
Sorry for being slow, but I have no idea how this code works. Where are the line directives? Are you sort of inferring them from the item locations? But then how do you deal with a line directive at the very end of the file, that is not followed by anything?

Reply:
A final line directive would be indeed be ignored. If we want to take them in account, we need to extract the information from the lexing buffer in the parsing function from `Pparse`. I am bit of wary of modifying those functions since they are commonly used function in the compiler libraries.

Reply:
But this is key, my point earlier was that the last code-includer-from-a-different-file standing would bracket its insertion with two line directives, the second going back to the "real" source file. If that inclusion is the last step in the user code, we do need the last line directive. Your implementation is a heuristic that is wrong in that case.